.github/workflows/run-tests.yml

```diff
@@ -47,3 +85,18 @@ jobs:
 
       - name: Execute tests
         run: composer test
+
+  check:
+    if: always()
+
+    needs:
+      - static-analysis
+      - test
+
+    runs-on: Ubuntu-latest
+
+    steps:
+      - name: Check if all jobs are passing
+        uses: re-actors/alls-green@release/v1
+        with:
+          jobs: ${{ toJSON(needs) }}
```

Comment:
@joelbutcher what is this doing exactly?

Reply:
@jbrooksuk Ah yes, sorry! I was supposed to add this into the PR description!

You should use this job as the "required check" that enables PR's in branch protection to determine if a PR is mergeable.

The way GitHub does this is weird when you have multiple CI pipelines, as well as using a matrix. GitHub doesn't mark the `check` job as `failed` it marks it as `skipped`.

The `if: always()` bit ensures this check is **always** run even if one of the `test` job matrix variants. or the `static-analysis` jobs fail.

Reply:
Love it!